app/apps/server/communication/uikit.js

```diff
@@ -83,6 +88,7 @@ export class AppUIKitInteractionApi {
 
 					const room = this.orch.getConverters().get('rooms').convertById(rid);
 					const user = this.orch.getConverters().get('users').convertToApp(req.user);
+					const visitor = this.orch.getConverters().get('visitors').convertByToken(req.visitor.token);
```

Comment:
since visitor is already found, you could use `convertVisitor`, right?

https://github.com/RocketChat/Rocket.Chat/blob/b5d199e74f5ea234097c9088725073d1442bfc27/app/apps/server/converters/visitors.js#L21

Reply:
Nope, the `visitor` from the middleware is missing some fields that the `findGuest` function doesn't include in the result. Since the fields are required in the app, we had to query again.

Another solution would be to use the converter directly in the middleware instead of the `findGuest` function, but it would differ from the approach done for the `user`.

Totally open for suggestions though :)

Reply:
I think we should avoid multiple round trips to DB, specially ones like this where we know it will always happen.. so either querying all needed fields during visitor validation, or not querying anything at all and just use `convertByToken` here.